gcs/src/gcs_group.cpp

```diff
@@ -1005,6 +1009,65 @@ group_recount_votes (gcs_group_t& group)
     return true;
 }
 
+/* Function to recompute vote based on only on the error code */
+int64_t recompute_vote_based_on_error_code (const gu::GTID& gtid,
+                                            const std::string& err_msg,
+                                            const int64_t old_vote,
+                                            const bool needs_logging)
+{
+
+    // Define a regular expression to match either a 4-digit or a 5-digit error
+    // code following "Error_code:"
+    std::regex errCodeRegex(R"(Error_code: (\b\d{4}\b|\b\d{5}\b))");
+
+    // A vector to store the matched 4-digit and 5-digit error codes
+    std::vector<std::string> matchedErrorCodes;
+
+    // Use std::sregex_iterator to find all matches in the error message
+    std::sregex_iterator it(err_msg.begin(), err_msg.end(), errCodeRegex);
+    std::sregex_iterator end;
+
+    // Iterate through all matches and store them in the vector
+    while (it != end)
+    {
+        // Extract the error code
+        std::smatch match = *it;
+        matchedErrorCodes.push_back(match[1].str());
+        ++it;
+    }
+
+    // Sort the matched error_codes in ascending order
+    std::sort(matchedErrorCodes.begin(), matchedErrorCodes.end());
+
+    // Create a stringstream to build the comma-separated string
+    std::stringstream ss;
+    for (size_t i = 0; i < matchedErrorCodes.size(); ++i)
+    {
+        ss << matchedErrorCodes[i];
+        if (i < matchedErrorCodes.size() - 1)
```

Comment:
I just realized that it doesn't matter if we have trailing coma or not, we even can join strings without any separator :)
But for the sake of pretty message printing, this is fine.